packages/core/test/oracle/OptimisticOracle.js

```diff
@@ -653,6 +653,16 @@ contract("OptimisticOracle", function (accounts) {
     assert.equal(await optimisticRequester.ancillaryData(), ancillaryData);
     await optimisticRequester.clearState();
 
+    // Check that OptimisticOracle stamped ancillary data as expected before sending to Oracle, and that we can decode
+    // it.
+    const priceRequests = await mockOracle.getPastEvents("PriceRequestAdded", { fromBlock: 0 });
+    assert.equal(priceRequests.length, 1, "should only be one price request escalated to MockOracle");
+    const stampedAncillaryData = priceRequests[0].returnValues.ancillaryData;
+    assert.equal(
+      hexToUtf8(stampedAncillaryData),
+      `${hexToUtf8(ancillaryData)},requester:${optimisticRequester.address.substr(2).toLowerCase()}`
```

Comment:
we'll probably want to add a common library method to parse ancilary data

Reply:
Agreed, that would be really nice. I don't think it would be too hard to write a parsing function that just dumps these key-value pairs into a js object. We can do this in a follow-up,